addons/hr_timesheet/hr_timesheet.py

```diff
@@ -103,129 +77,118 @@ def on_change_unit_amount(self, cr, uid, id, prod_id, unit_amount, company_id, u
             res['value'].update({'product_uom_id': False})
         return res
 
-    def _getEmployeeProduct(self, cr, uid, context=None):
-        if context is None:
-            context = {}
-        emp_obj = self.pool.get('hr.employee')
-        emp_id = emp_obj.search(cr, uid, [('user_id', '=', context.get('user_id') or uid)], context=context)
-        if emp_id:
-            emp = emp_obj.browse(cr, uid, emp_id[0], context=context)
-            if emp.product_id:
-                return emp.product_id.id
-        return False
-
-    def _getEmployeeUnit(self, cr, uid, context=None):
-        emp_obj = self.pool.get('hr.employee')
-        if context is None:
-            context = {}
-        emp_id = emp_obj.search(cr, uid, [('user_id', '=', context.get('user_id') or uid)], context=context)
-        if emp_id:
-            emp = emp_obj.browse(cr, uid, emp_id[0], context=context)
-            if emp.product_id:
-                return emp.product_id.uom_id.id
-        return False
-
-    def _getGeneralAccount(self, cr, uid, context=None):
-        emp_obj = self.pool.get('hr.employee')
-        if context is None:
-            context = {}
-        emp_id = emp_obj.search(cr, uid, [('user_id', '=', context.get('user_id') or uid)], context=context)
-        if emp_id:
-            emp = emp_obj.browse(cr, uid, emp_id[0], context=context)
-            if bool(emp.product_id):
-                a = emp.product_id.property_account_expense.id
-                if not a:
-                    a = emp.product_id.categ_id.property_account_expense_categ.id
-                if a:
-                    return a
-        return False
-
-    def _getAnalyticJournal(self, cr, uid, context=None):
-        emp_obj = self.pool.get('hr.employee')
-        if context is None:
-            context = {}
-        if context.get('employee_id'):
-            emp_id = [context.get('employee_id')]
+    @api.model
+    def _get_employee_product(self, user_id=None):
+        emp = self.env['hr.employee'].search([('user_id', '=', self.user_id.id or user_id  or self.env.uid)])
+        if emp and emp[0].product_id.id:
```

Comment:
You can directly use emp.product_id.id (takes the first element of the record set by default).